airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -139,3 +139,13 @@ def _get_aws_credentials(self) -> Tuple[Optional[Any], Optional[Any]]:
 
     def set_autocommit(self, conn, autocommit: Any) -> None:
         conn.autocommit(autocommit)
+
+    def run(self, sql, autocommit=False, parameters=None):
+        """Snowflake-connector doesn't allow natively the execution of multiple SQL statements in the same call. So for
+        allowing to pass files or strings with several queries this method is coded, that relies on run from DBApiHook"""
+        if isinstance(sql, str):
+            queries = sql.split(';')
```

Comment:
Is it backward compatible?
What happens when user has single query that search text column for substring `;` as char? I think this requires to add note about it.

What do you think about exposing `execute_string` which can handle multiple SQLs in single string in addition to the run method of dbapi hook.

https://docs.snowflake.com/en/user-guide/python-connector-api.html#execute_string

https://github.com/snowflakedb/snowflake-connector-python/blob/master/src/snowflake/connector/connection.py#L493

Reply:
A query that searches for ';' has been a concern to me. I don't know how often is that, though, but I think I could add some parameter or something to make a different behavior in that case (based in re, for instance)

I'll take a look at execute_string

Reply:
I have updated the code using execute_string and adding the previous method for not breaking backwards compatibility in case you were using a list of queries